widgetastic_manageiq/__init__.py

```diff
@@ -4051,3 +4051,461 @@ def data_for_timestamp(self, timestamp):
+    next_btn = Button('Next')
+    cancel_btn = Button('Cancel')
+
+
+class InfraMappingCommonButtonsSet2(View):
+
+    add_mapping = Button('Add Mapping')
+    remove_mapping = Button('Remove Selected')
+    remove_all_mappings = Button('Remove All')
+    mappings_tree = InfraMappingTreeView(tree_class='treeview')
+
+
+class InfraMappingWizardGeneralView(InfraMappingCommonButtonsSet1):
+    name = TextInput(name='name')
+    description = TextInput(name='description')
+
+    def fill(self, values):
+        try:
+            self.name.fill(values['name'])
+            self.description.fill(values['description'])
+            return True
+        except Exception as e:
+            self.logger.debug("Following Exception occured while trying to fill:{}".format(e))
+            return False
+
+
+class InfraMappingWizardClustersView(InfraMappingCommonButtonsSet1, InfraMappingCommonButtonsSet2):
+    source_clusters = MultiSelectBox('source_clusters')
+    target_clusters = MultiSelectBox('target_clusters')
+
+    @property
+    def is_displayed(self):
+        return self.source_clusters.item_count > 0 or self.target_clusters.item_count > 0
+
+    def fill(self, values, partial_match=False):
+        """Use _add_cluster_mapping to add all mappings specified in values.
+
+        Args:
+        values:
+            format as following:
+                {
+                    'mapping1':
+                    {
+                        'sources':['item1','item2'],
+                        'target': 'item_target1'
+                    },
+                    'mapping2':
+                    {
+                        'sources':['item3','item4'],
+                        'target': 'item_target2'
+                    },
+                    ...
+                }
+        """
+        try:
+            for mapping in values:
+                self._add_cluster_mapping(values[mapping]['sources'], values[mapping]['target'],
+                    partial_match=partial_match)
+            return True
+        except Exception as e:
+            self.logger.debug("Following Exception occured while trying to fill:{}".format(e))
+            return False
+
+    def _add_cluster_mapping(self, sources=[], target=None, partial_match=False):
+        """Add mapping from one or more sources to one target.
+
+        Args:
+            sources: list of source items
+            target: str containing target item
+        """
+
+        if isinstance(sources, list) and sources and target:
+            for item in sources:
+                self.logger.info("Mapping Source Cluster {} to Target Cluster {}"
+                    .format(item, target))
+                self.source_clusters.select_by_text(item, partial_match=partial_match)
+                self.target_clusters.select_by_text(target, partial_match=partial_match)
+                self.add_mapping.click()
+
+
+class InfraMappingWizardDatastoresViw(InfraMappingCommonButtonsSet1, InfraMappingCommonButtonsSet2):
```

Comment:
**REQUIRED**
1. Typo is here. 
2. Don't inherit `InfraMappingCommonButtonsSet1` and `InfraMappingCommonButtonsSet2`, but include them using `View.include`

Reply:
I'd agree @quarckster, InfraMappingCommonButtonsSet1/2 being base classes for other V2V widgets creates a bit of an odd structure. `View.include` is perfect here.